### PR TITLE
ssh tasks: default and console

### PR DESCRIPTION
**Pull request by cainlevy**
Added support for:

* `cap $environment ssh`: opens a remote shell on a random server
* `cap $environment console:$role`: opens a remote Rails console on a server with the given $role
* `cap $environment console`: opens a remote Rails console on a random server

### CODE REVIEW

**Comment by devpuppy** [comment]
I've trained myself to do `cap $environment ssh:bg` so that I don't take up memory on a server that doesn't have enough to spare. Is this no longer an issue?

Or maybe add a `cap $enviornment console:$role` command as well for better targeting?

**Comment by cainlevy** [comment]
@devpuppy i could update this to prefer bg servers. i started down this path because i was working in a highlander environment that didn't have any of the default roles!

**Comment by cainlevy** [comment]
oh, actually, we can't hard-code `bg` into this repo since roles are a matter of project config. so yeah, adding `cap $environment console:$role` makes a lot of sense.

**Comment by cainlevy** [comment]
updated

**Comment by blinsay** [comment]
How hard would it be to only allow `cap $env ssh` if all of the roles map to the same server?

(I don't even know if that's useful. Just stirring the pot)

File: lib/capistrano/hivequeen/capistrano_configuration.rb [hunk 1]
```diff
@@ -88,19 +88,39 @@
...
+    HiveQueen.default_roles.each do |role_name|
+      task role_name do
+        puts "Opening console"
+        exec command_format % [user, roles[role_name.to_sym].servers.first]
+      end
+    end
+
+    task :default do
+      puts "Opening console"
+      exec command_format % [user, roles.values.sample.servers.sample]
```
Comment (ktheory) [inline lib/capistrano/hivequeen/capistrano_configuration.rb]: How about using `bg` role by default? Wouldn't want a resource-intensive ruby console to affect web/app roles.

Comment (ktheory) [inline lib/capistrano/hivequeen/capistrano_configuration.rb]: D'oh! Just catching up on the other comments. :wink:

**Comment by ktheory** [comment]
LGTM.

**Comment by samuelcole** [comment]
:wave:

**Comment by ktheory** [comment]
@samuelcole !!! :heart: :heart: :heart:

**Comment by cainlevy** [comment]
edited so that role-based tasks (e.g. `cap $env ssh:$role`) are defined once we know which roles are actually present.

**Comment by cainlevy** [comment]
i think this is good! planning to merge tomorrow pending feedback.

> How hard would it be to only allow cap $env ssh if all of the roles map to the same server?

probably doable. gonna wait until we have reason to need it. meanwhile i'm happy to assume that folks will target a role when appropriate.

**Comment by blinsay** [comment]
> meanwhile i'm happy to assume that folks will target a role when appropriate

:+1: :+1:

**Pull request merged**